core/qtserialbluetooth.cpp

```diff
@@ -106,14 +109,15 @@ static dc_status_t qt_serial_open(qt_serial_t **io, dc_context_t*, const char* d
 
 	// Wait until the connection succeeds or until an error occurs
 	QEventLoop loop;
-	loop.connect(serial_port->socket, SIGNAL(connected()), SLOT(quit()));
-	loop.connect(serial_port->socket, SIGNAL(error(QBluetoothSocket::SocketError)), SLOT(quit()));
+	QObject::connect(serial_port->socket, &QBluetoothSocket::connected, &loop, &QEventLoop::quit);
+	QObject::connect(serial_port->socket, QOverload<QBluetoothSocket::SocketError>::of(&QBluetoothSocket::error),
+		&loop, [&loop](QBluetoothSocket::SocketError) { loop.quit(); });
```

Comment:
Here we reached the point where I KNOW what the old code does and I have not the faintest clue what the new code does.
Sure, you know a lot more about Qt and C++ programming than I do, but for now I assume that I'll have to maintain this syntax salad.
Ugh.

Reply:
The second part is luckily not needed at all, one can directly connect here.